eclipse-theia/theia-cloud
# GH-349: Extend service with a list app definitions endpoint

## Changes

### App definition endpoint
- Add AppDefinitionResource with endpoint to list all app definitions. The endpoint is restricted to authenticated users but is still available in anonymous mode

### Sensitive data redaction
- Add `SensitiveData` annotation to mark properties that should not be serialized publicly by Jackson.
- Add a corresponding serializer and serializer modifiert for Jackson and register the modifier in the service
- Add unit tests for the serializer

### Javascript API
- Update openapi.json from service
- Regenerate api code
- Add AppDefinitions namespace with function to list app definitions

### Testing Page
- Add button to get app definitions
- Apply formatting rules to App.tsx by saving it
- Minor fix in example keycloak URL to remove obsolete `/auth`

### Misc
- Add mockito dependency to the common maven module

## How to test

- Run service either by building it in Minikube or debugging it
- Start testing page. Use the List App Definitions button to see the app definitions
- Ensure the `pullSecret` properties have value `***` which is the redacted value
